pkg/component/extensions/operatingsystemconfig/original/components/containerd/initializer.go

```diff
@@ -33,15 +33,17 @@ func (initializer) Name() string {
 	return "containerd-initializer"
 }
 
-func (initializer) Config(ctx components.Context) ([]extensionsv1alpha1.Unit, []extensionsv1alpha1.File, error) {
-	const (
-		pathScript          = v1beta1constants.OperatingSystemConfigFilePathBinaries + "/init-containerd"
-		unitNameInitializer = "containerd-initializer.service"
-	)
+const (
+	// InitializerScriptPath is the path of the containerd initializer script.
+	InitializerScriptPath = v1beta1constants.OperatingSystemConfigFilePathBinaries + "/init-containerd"
+	// InitializerUnitName is the name of the containerd initializer service.
+	InitializerUnitName = "containerd-initializer.service"
+)
 
+func (initializer) Config(_ components.Context) ([]extensionsv1alpha1.Unit, []extensionsv1alpha1.File, error) {
```

Comment:
> On new nodes, there is no earlier defined dropin
on containerd. The deprecated `containerd-initializer`
can be skipped safely.

I am not able to understand how the drop-in is created for old Nodes but no longer created for new Nodes. os-gardenlinux@master still has https://github.com/gardener/gardener-extension-os-gardenlinux/blob/a2debf66f59d4d163f53fbfcf6fe3b0911ec3805/pkg/controller/operatingsystemconfig/actuator.go#L81.

Reply:
The dropin on old nodes is already there, when we roll-out the change of this PR. This is the reason, we can't simply remove the `containerd-initializer` unit. Extensions might place(d) additional dropins, so they have to do it similarly.